sdk/src/azure/core/az_mqtt5_request_hfsm.c

```diff
@@ -108,9 +108,22 @@ static az_result root(az_event_policy* me, az_event event)
     case AZ_HFSM_EVENT_ERROR:
     {
       log_event(event, root_string, this_policy->_internal.correlation_id);
-      if (az_result_failed(az_event_policy_send_inbound_event(me, event)))
+      az_hfsm_event_data_error* error_data = (az_hfsm_event_data_error*)event.data;
+      if (error_data->error_type == AZ_MQTT5_EVENT_RPC_CLIENT_ERROR_RSP)
       {
-        az_platform_critical_error();
+        // if the RPC Client goes to faulted, fault all requests so they can be cleaned up.
+        az_mqtt5_rpc_client_rsp_event_data resp_data
+            = { .response_payload = AZ_SPAN_EMPTY,
+                .status = AZ_MQTT5_RPC_STATUS_UNKNOWN,
+                .error_message = AZ_SPAN_FROM_STR("RPC Client Faulted."),
+                .content_type = AZ_SPAN_EMPTY,
+                .correlation_id = this_policy->_internal.correlation_id };
+
+        _az_CRITICAL_IF_FAILED(_az_hfsm_transition_substate((_az_hfsm*)me, root, faulted));
```

Comment:
I don't think this transition is reflected on the Request puml

Reply:
Wasn't 100% sure how to signify this was from root, but it's captured in the puml!
![image](https://github.com/Azure/azure-sdk-for-c/assets/54871851/3e456678-5ae4-4111-8d43-75c2620116cc)